File: 프로그래머스/Level2/N개의 최소공배수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> arr) {
    
    sort(arr.begin(), arr.end(), less<int>());
    int i = 1;
    int max = arr[arr.size()-1];
    
    vector<int> num;
    
    while(1){
        int cnt = 0;
        for(int j=0; j<arr.size(); j++){
            if((max*i)%arr[j] == 0){
                cnt ++;
            }
        }
        if(cnt == arr.size()) return max*i;
        i++;
    }
}

File: 프로그래머스/Level2/N개의 최소공배수.js
function solution(arr) {
    arr = arr.sort((a, b) => b-a);
    
    let i = 1;
    
    while(1){
        const LCM = arr[0] * i;
        const tmp = arr.filter(item => LCM%item === 0);
        
        if(tmp.length === arr.length) return LCM;
        else i++;
    }
}
